src/main/java/com/ibm/disni/RdmaEndpoint.java

```diff
@@ -112,7 +112,8 @@ public synchronized void connect(SocketAddress dst, int timeout) throws Exceptio
 		if (connState != CONN_STATE_ROUTE_RESOLVED){
 			throw new IOException("resolve route failed");
 		}			
-		
+
+		group.checkResources(idPriv);
```

Comment:
Why are we funneling this through the group, why not calling idPriv checkresources directly from within the endpoint?